### PR TITLE
docs: update with LogUp + range checker changes

### PR DESCRIPTION
**Pull request by grjte**
This PR updates the docs for the new range checker construction which makes lookup of the LogUp lookup argument.

- [x] generalize the lookup argument docs, adding new pages for overview and logup and reducing contents of the page on multiset checks
- [x] update range checker construction
- [x] update all component docs which refer to lookup constructions
- [x] update & create new excalidraw images
- [x] read through & double-check all content & links

### CODE REVIEW

File: docs/src/design/lookups/main.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+These specialized components are designed to prove the internal correctness of the execution of the operations they support. However, in isolation they cannot make any guarantees about the source of the input data or the destination of the output data.
+
+In order to prove that the inputs and outputs specified by the main circuit match the inputs and outputs provably executed in the specialized component, some kind of provable communication bus is needed.
+
+This bus is typically implemented as some kind of lookup argument, and in Miden VM in particular we use multiset checks or LogUp.
+
+Miden VM uses 2 communication buses:
+
+- The chiplets bus [$b_{chip}$](../chiplets/main.md#chiplets-bus), which communicates with all of the chiplets (Hash, Bitwise, and Memory). It is implemented using multiset checks.
```
Comment (bobbinth) [inline docs/src/design/lookups/main.md:42]: nit: we should probably mention Kernel ROM chiplet here too.

File: docs/src/design/lookups/logup.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+\sum_{i=0}^{l} \frac{1}{(\alpha - a_i)} = \sum_{i=0}^{n} \frac{m_i}{(\alpha - b_i)}
+$$
+
+In the above:
+- $l$ is the number of values in $a$
+- $n$ is the number of values in $b$
+- $m_i$ is the multiplicity of $b_i$, which is expected to match the number of times the value $b_i$ is duplicated in column $a$
+- $\alpha$ is a random value that is sent to the prover by the verifier after the prover commits to the execution trace of the program.
+
+Thus, instead of needing to compute running products, we are able to assert correct lookups by computing running sums.
```
Comment (bobbinth) [inline docs/src/design/lookups/logup.md:17]: I believe there is a limitation on the value of multiplicity - but I don't remember what it is exactly. We should probably mention it here.

**Review by bobbinth — APPROVED** [review]
Looks great! Thank you! I left a couple minor comments inline - but we can merge even without them.

File: docs/src/design/lookups/logup.md (hunk 2 above)
Comment (grjte) [reply docs/src/design/lookups/logup.md:17]: I was thinking the multiplicity was limited by the size of the field, which I stated somewhere. Is there another limitation?

File: docs/src/design/lookups/main.md (hunk 1 above)
Comment (grjte) [reply docs/src/design/lookups/main.md:42]: Good point. I can update this and address the multiplicity comment together once I know your thought on that one. If you want to merge this first that's also fine & I can address both points in a small follow-up PR

File: docs/src/design/lookups/logup.md (hunk 2 above)
Comment (bobbinth) [reply docs/src/design/lookups/logup.md:17]: It's either that or the size of the domain (or I wonder if having multiplicity bigger than the size of the domain causes some kind of security degradation).

Comment (Al-Kindi-0) [reply docs/src/design/lookups/logup.md:17]: Both $l$ and $n$ should be smaller than $p$ the size of field, from the paper. The multiplicities should be smaller than the size of the set we are looking up into, that's my understanding from what I recall about the paper.

Comment (grjte) [reply docs/src/design/lookups/logup.md]: I've updated this section, as well as at `docs/src/design/range.md` line 62,

**Pull request merged**